woocommerce/woocommerce
# Set up and migration to use BlockTemplate classes

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR moves our simple product template to the new `BlockTemplate` classes and creates the necessary interfaces and classes for the `SimpleProductTemplate`.

Closes #39421 and #39515  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Load two WooCommerce websites, one with the latest of `trunk` and one with this branch, both with the **Product Block Editor** enabled under the **Advanced > Features** settings.
2. Open the **Product > Add New** for each website
3. Walk through each tab, starting with `General` ending with `Shipping` ( bonus points for checking `Variations` as well, which requires the `product-variation-management` feature flag to be enabled using the WC Test Helper ).
4. Check if the tabs, sections and fields are the same and in the same order. Everything should work the same as before.
5. Run new PHP Unit tests:
```
pnpm --filter=woocommerce env:test
pnpm test:unit:env --filter 'Automattic\\WooCommerce\\Tests\\Admin\\ProductBlockEditor\\ProductTemplates'
```



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
